### PR TITLE
Adding v0.5.0 blog post

### PR DESCRIPTION
**Pull request by robscott**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
This is a follow up to @shaneutt's PR: #1227 and mirrors https://github.com/kubernetes/website/pull/34968.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

/hold for release

### CODE REVIEW

File: site-src/blog/index.md [hunk 1]
```diff
@@ -1,5 +1,22 @@
...
+
+We are excited to announce the v0.5.0 release of Gateway API. For the first
+time, several of our most important Gateway API resources are graduating to
+beta. Additional, we are starting a new initiative to explore how Gateway API
+can be used for mesh and introducing new experimental concepts such as URL
+rewrites.
+
+[:octicons-arrow-right-24: Continue reading][Gateway API Graduates to Beta]
+
+[Gateway API Graduates to Beta]:/blog/2022/graduating-to-beta
```
Comment (sftim) [inline site-src/blog/index.md:18]: This URL doesn't match the one seen in https://github.com/kubernetes/website/blame/6d012150bf99db65f41408eea8471efc21be1034/content/en/blog/_posts/2022-07-13-gateway-api-in-beta.md#L6

(https://gateway-api.sigs.k8s.io/blog/2022/introducing-v1beta1/)

Let's make sure these match up.

**Review by sftim — CHANGES_REQUESTED** [review]

File: mkdocs.yml [hunk 2]
```diff
@@ -69,5 +69,7 @@ nav:
     - GAMMA: contributing/gamma.md
   - Blog: 
     - Index: blog/index.md
+    - 2022:
+      - blog/2022/first-beta.md
```
Comment (sftim) [inline mkdocs.yml:73]: URL path `blog/2022/first-beta.md` ≭ repo path `site-src/blog/2022/graduating-to-beta.md`

File: site-src/blog/index.md (hunk 1 above)
Comment (robscott) [reply site-src/blog/index.md:18]: Woops, updated in both places, thanks!

File: mkdocs.yml (hunk 2 above)
Comment (robscott) [reply mkdocs.yml]: Updated, thanks!

**Comment by sftim** [comment]
Is this ready to go out? Looks that way.

**Comment by sftim** [comment]
@robscott (or another person from the Gateway project, eg @youngnick): the next steps, as I see them:

1. **do the release**
   - make sure that https://github.com/kubernetes-sigs/gateway-api/releases/tag/v0.5.0 is live
   - make sure that https://gateway-api.sigs.k8s.io/guides/getting-started/ links to the v0.5.0 release
1. unhold this PR **once ready**
1. wait for https://gateway-api.sigs.k8s.io/blog/2022/graduating-to-beta/ to be live
   - and check it looks right
1. unhold https://github.com/kubernetes/website/pull/34968

**Comment by robscott** [comment]
Thanks @sftim! We're hoping to have the release out in around an hour and then we'll remove the holds on these blog posts in the order suggested above.

**Review by shaneutt — APPROVED** [review]
/lgtm

**Pull request merged**